src/components/DiagnosticReport/DiagnosticReportComponent.js

```diff
@@ -42,38 +42,33 @@ const DiagnosticReportComponent = ({ data, consentReqId }) => {
 
   function getResultsList(results, resourceType) {
     const referenceList = [];
-    results
-      ? results.map((result) => {
-        const resource = result.targetResource;
-        referenceList.push(resource);
-      })
-      : undefined;
-    return referenceList.filter((ref) => (ref.resourceType = resourceType));
+    if (results) results.forEach((result) => referenceList.push(result.targetResource));
+    return referenceList.filter((ref) => (ref.resourceType === resourceType));
```

Comment:
Looks like there is some change in logic here.
It was `(ref) => (ref.resourceType = resourceType)`, which looks wrong. I am wondering how it was working before.

Reply:
I don't think filtering was working before. Can you please check on dev?

Reply:
This looks correct.